### PR TITLE
Gsf5 fixes

### PR DESCRIPTION
**Pull request by MarekM25**
Fix issues that we noticed during transition on GSF5

### CODE REVIEW

File: src/Nethermind/Nethermind.Merge.Plugin/Handlers/V1/NewPayloadV1Handler.cs [hunk 1]
```diff
@@ -143,8 +143,19 @@ public async Task<ResultWrapper<PayloadStatusV1>> HandleAsync(ExecutionPayloadV1
...
             }
 
-            bool parentProcessed = _blockTree.WasProcessed(parentHeader.Number, parentHeader.GetOrCalculateHash());
-            if (!parentProcessed)
+            BlockInfo parentBlockInfo = _blockTree.GetInfo(parentHeader.Number, parentHeader.GetOrCalculateHash()).Info;
+            bool parentProcessed = parentBlockInfo.WasProcessed;
+
+            // edge-case detected on GSF5 - during the transition we want to try process all transition blocks from CL client
+            // The last condition: !parentBlockInfo.IsBeaconBody will be true for terminal blocks. Checking _posSwitcher.IsTerminal might not be the best, because we're loading parentHeader with DoNotCalculateTotalDifficulty option
+            bool forceProcessing = !_poSSwitcher.TransitionFinished && (_blockTree.Head?.Number ?? 0) + 8 >= block.Number && !parentBlockInfo.IsBeaconBody;
```
Comment (asdacap) [inline src/Nethermind/Nethermind.Merge.Plugin/Handlers/V1/NewPayloadV1Handler.cs:151]: 8?

File: src/Nethermind/Nethermind.Merge.Plugin/MergeSealValidator.cs [hunk 2]
```diff
@@ -32,9 +32,9 @@ public class MergeSealValidator : ISealValidator
...
         _preMergeSealValidator = preMergeSealValidator;
     }
-    public bool ValidateParams(BlockHeader parent, BlockHeader header) => 
+    public bool ValidateParams(BlockHeader parent, BlockHeader header) =>
         _poSSwitcher.IsPostMerge(header) || _preMergeSealValidator.ValidateParams(parent, header);
 
-    public bool ValidateSeal(BlockHeader header, bool force) => 
-        _poSSwitcher.IsPostMerge(header) || _preMergeSealValidator.ValidateSeal(header, force);
+    public bool ValidateSeal(BlockHeader header, bool force) =>
+        _poSSwitcher.GetBlockConsensusInfo(header, true).IsPostMerge || _preMergeSealValidator.ValidateSeal(header, force);
```
Comment (asdacap) [inline src/Nethermind/Nethermind.Merge.Plugin/MergeSealValidator.cs:39]: qq: why we don't check the seal if it is post merge?

File: src/Nethermind/Nethermind.Synchronization/SyncServer.cs [hunk 3]
```diff
@@ -180,7 +180,8 @@ public void AddNewBlock(Block block, ISyncPeer nodeWhoSentTheBlock)
 
             bool isBlockBeforeTheSyncPivot = block.Number < _pivotNumber;
             bool isBlockOlderThanMaxReorgAllows = block.Number < (_blockTree.Head?.Number ?? 0) - Sync.MaxReorgLength;
-            bool isBlockTotalDifficultyLow = block.TotalDifficulty < _blockTree.BestSuggestedHeader.TotalDifficulty;
+            bool isBlockTotalDifficultyLow = block.TotalDifficulty < _blockTree.BestSuggestedHeader.TotalDifficulty
+                                             && (_specProvider.TerminalTotalDifficulty == null || block.TotalDifficulty < _specProvider.TerminalTotalDifficulty); // terminal blocks with lower TTD might be useful for smooth merge transition
```
Comment (asdacap) [inline src/Nethermind/Nethermind.Synchronization/SyncServer.cs:184]: What if there are two terminal block and the lower one appear later? I assume it must go through NP?

Comment (MarekM25) [reply src/Nethermind/Nethermind.Synchronization/SyncServer.cs:184]: If terminal block appears later that shouldn't be block choosen by CL clients to build chain on top of it. However, in this case we will get NP(childOfTerminalBlock) we will return SYNCING and we should be able to catch up with MergeBlockDownloader

File: src/Nethermind/Nethermind.Merge.Plugin/MergeSealValidator.cs (hunk 2 above)
Comment (MarekM25) [reply src/Nethermind/Nethermind.Merge.Plugin/MergeSealValidator.cs:39]: It is PoW specific thing, so we don't have it post merge

File: src/Nethermind/Nethermind.Merge.Plugin/Handlers/V1/NewPayloadV1Handler.cs (hunk 1 above)
Comment (MarekM25) [reply src/Nethermind/Nethermind.Merge.Plugin/Handlers/V1/NewPayloadV1Handler.cs:151]: We're using 8 here too: https://github.com/NethermindEth/nethermind/blob/master/src/Nethermind/Nethermind.Facade/Eth/EthSyncingInfo.cs#L37

I should change this condition bool weAreCloseToHead = (_blockTree.Head?.Number ?? 0) + 8 >= block.Number;

File: src/Nethermind/Nethermind.Merge.Plugin/MergeSealValidator.cs (hunk 2 above)
Comment (asdacap) [reply src/Nethermind/Nethermind.Merge.Plugin/MergeSealValidator.cs:39]: Isnt seal mean the hash?

Comment (asdacap) [reply src/Nethermind/Nethermind.Merge.Plugin/MergeSealValidator.cs:39]: Isnt seal mean the hash?

**Review by asdacap — APPROVED** [review]

File: src/Nethermind/Nethermind.Merge.Plugin/MergeSealValidator.cs (hunk 2 above)
Comment (MarekM25) [reply src/Nethermind/Nethermind.Merge.Plugin/MergeSealValidator.cs:39]: https://github.com/NethermindEth/nethermind/blob/master/src/Nethermind/Nethermind.Consensus.Ethash/EthashSealValidator.cs#L77

**Comment by MarekM25** [comment]
The workflow is not starting the tests, but I run them locally.

**Pull request merged**